arch/xtensa/src/common/xtensa_initialstate.c

```diff
@@ -100,6 +124,71 @@ void up_initial_state(struct tcb_s *tcb)
   xcp->regs[REG_A1] = (uint32_t)tcb->stack_base_ptr + /* Physical top of stack frame    */
                                 tcb->adj_stack_size;
 
+  /* Each task access the TLS variables using the THREADPTR register plus an
+   * offset to obtain the address of the variable.
+   *
+   * TLS layout at link-time, where 0xNNN is the offset that the linker
+   * calculates to a particular TLS variable.
```

Comment:
is this a modified copy of https://github.com/espressif/esp-idf/blob/b5ac4fbdf9e9fb320bb0a98ee4fbaa18f8566f37/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c#L218 ?

Reply:
Yes, but the explanation is not well documented (and there are some changes regarding the position of the TLS data on stack.